core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -108,28 +109,32 @@
     }
 
     public Environment(Settings settings) {
+        this(PathUtils.getDefaultFileSystem(), settings);
+    }
+
+    public Environment(FileSystem fs, Settings settings) {
```

Comment:
I still dont' think it is necessary. The mock FS needs to be installed before creating the Environemnt, and any calls to PathUtils.get will work correctly, including the one done for tmpDir.